OCR documentation (#634)
JabRef/user-documentation: en/advanced/OCR.md

# OCR

> OCR (Optical Character Recognition) is a technology that is used to convert scanned PDFs into editable and searchable data. OCR can be used via multiple tools and engines. Currently, JabRef provides OCR using [OCRmyPDF](https://ocrmypdf.readthedocs.io/en/latest/) and more engine are coming soon.

Reviewer: ```suggestion
[OCR](https://en.wikipedia.org/wiki/Optical_character_recognition) (Optical Character Recognition) is defined as the electronic or mechanic conversion of images of typed, handwritten or printed text into machine-encoded text. In other words, it is a technique that adds editable and searchable data to PDFs and other files in your library. OCR can be used via multiple tools and engines. Currently, JabRef provides OCR using [OCRmyPDF](https://ocrmypdf.readthedocs.io/en/latest/).
```
Only document what is, not what will be, because plans may change.

> ## OCRmyPDF

Reviewer: Split into two sections.

1. How to install OCRmyPDF
  - add a link to their website
  - add a link to their installation documentation
2. How to perform OCR

Once more engines are added, OCRmyPDF and the others could become a subsection of the "How to perform OCR" section, but let's leave that to the future us.

{% hint style="warning" %}

> OCRmyPDF must be installed on your system before using this feature. See the installation instructions below.

Reviewer: ```suggestion
OCRmyPDF must be installed on your system to use this feature.
```

Author: I provided the installation link but seems to be removed by accident, should I add it again?

Author: I think I clicked ctrl+z by accident.

Reviewer: I wouldn't put the link into the warning. Yeah, I would add the link in a section down below.

# OCR

> OCR (Optical Character Recognition) is defined as the electronic or mechanic conversion of images of typed, handwritten or printed text into machine-encoded text. In other words, it is a technique that adds editable and searchable data to PDFs and other files in your library. OCR can be used via multiple tools and engines. Currently, JabRef provides OCR using [OCRmyPDF](https://ocrmypdf.readthedocs.io/en/latest/).

Reviewer: ```suggestion
[OCR](https://en.wikipedia.org/wiki/Optical_character_recognition) (Optical Character Recognition) is defined as the electronic or mechanic conversion of images of typed, handwritten or printed text into machine-encoded text. In other words, it is a technique that adds editable and searchable data to PDFs and other files in your library. OCR can be used via multiple tools and engines. Currently, JabRef provides OCR using [OCRmyPDF](https://ocrmypdf.readthedocs.io/en/latest/).
```

## OCRmyPDF

{% hint style="warning" %}
OCRmyPDF must be installed on your system to use this feature. See the installation instructions below.
{% endhint %}

> 

Reviewer: I missed that. I think the warning should be under the how to perform OCR section and the OCRmyPDF headline should be removed. Then turn the other two sections from level three into level two headers.

> 1. How to install [OCRmyPDF](https://github.com/ocrmypdf/ocrmypdf?tab=readme-ov-file)

Reviewer: ```suggestion
### How to install [OCRmyPDF](https://github.com/ocrmypdf/ocrmypdf)
```
Headers like `###` are useful to websearch engines.

>    * Check [installation guide](https://ocrmypdf.readthedocs.io/en/latest/installation.html) according to your operating system.

Reviewer: ```suggestion
   * Please check the [installation guide](https://ocrmypdf.readthedocs.io/en/latest/installation.html) and follow the instructions for your operating system.
```

> 2. How to perform OCR on a scanned PDF file in JabRef

Reviewer: ```suggestion
### How to perform OCR on a scanned PDF file in JabRef
```

* After performing OCR, JabRef creates a new PDF file with the OCR text embedded, and it will be linked in all the entries that have the old file linked to them. The original scanned PDF will remain unchanged.

    ![original-and-ocred-files.png](original-and-ocred-files.png)

* Now you can select and search text in the new PDF file.

    ![comparison-between-original-and-ocred-file.png](comparison-between-original-and-ocred-file.png)

> {% hint style="warning" %}

Reviewer: Ah, lol. Sorry, my bad. I used unclear language. I meant not directly under, but rather beneath the header, but above the body.

Author: Ah ok no problem
